lib/index.js

```diff
@@ -236,7 +240,11 @@ function tryCatch (tryFn, catchFn) {
 function apiExtend (tree) {
   tree.walk = api.walk
   tree.match = api.match
-  tree.messages = api.messages
+  // BUG previous messages aren't cleaned up
```

Comment:
I'm either doing something wrong here or we need to change this to not be extended by the API, since this also causes issue in `html-loader@1.0.0-alpha` where I need to manually reset the messages between recursive runs atm

Reply:
Maybe we can still extend it via the API and need to make a change elsewhere, not sure yet 🙃

Reply:
i see after move to branch )